snap/plugin/snap-collector-voxter/voxter/voxter.go

```diff
@@ -0,0 +1,151 @@
+package voxter
+
+import (
+	"fmt"
+	"time"
+	"strings"
+
+	"github.com/gosimple/slug"
+	. "github.com/intelsdi-x/snap-plugin-utilities/logger"
+	"github.com/intelsdi-x/snap/control/plugin"
+	"github.com/intelsdi-x/snap/control/plugin/cpolicy"
+	"github.com/intelsdi-x/snap/core"
+	"github.com/intelsdi-x/snap/core/ctypes"
+)
+
+const (
+	// Name of plugin
+	Name = "voxter"
+	// Version of plugin
+	Version = 1
+	// Type of plugin
+	Type = plugin.CollectorPluginType
+	// stat url
+	statsURL = "https://vortex2.voxter.com/api/"
+)
+
+var (
+	statusMap = map[string]int{"up": 0, "down": 1}
+)
+
+func init() {
+	slug.CustomSub = map[string]string{".": "_"}
+}
+
+// make sure that we actually satisify required interface
+var _ plugin.CollectorPlugin = (*Voxter)(nil)
+
+type Voxter struct {
+}
+
+// CollectMetrics collects metrics for testing
+func (v *Voxter) CollectMetrics(mts []plugin.MetricType) ([]plugin.MetricType, error) {
+	var err error
+	metrics := make([]plugin.MetricType, 0)
+	conf := mts[0].Config().Table()
+	apiKey, ok := conf["voxter_key"]
+	if !ok || apiKey.(ctypes.ConfigValueStr).Value == "" {
+		LogError("voxter_key missing from config.")
+		return nil, fmt.Errorf("voxter_key missing from config, %v", conf)
+	}
+	client, err := NewClient(statsURL, apiKey.(ctypes.ConfigValueStr).Value, false)
+	if err != nil {
+		LogError("failed to create voxter api client.", "error", err)
+		return nil, err
+	}
+	LogDebug("request to collect metrics", "metric_count", len(mts))
+
+	resp, err := v.EndpointMetrics(client, mts)
+	if err != nil {
+		LogError("failed to collect metrics.", "error", err)
+		return nil, err
+	}
+	metrics = resp
+
+	LogDebug("collecting metrics completed", "metric_count", len(metrics))
+	return metrics, nil
+}
+
+//GetMetricTypes returns metric types for testing
+func (v *Voxter) GetMetricTypes(cfg plugin.ConfigType) ([]plugin.MetricType, error) {
+	mts := []plugin.MetricType{}
+
+	mts = append(mts, plugin.MetricType{
+		Namespace_: core.NewNamespace("raintank", "apps", "voxter", "endpoints", "*", "*", "registrations"),
```

Comment:
please use the DynamicElements instead of just *

eg. https://github.com/raintank/raintank-apps/blob/master/snap/plugin/snap-collector-rt-gitstats/gitstats/gitstats.go#L321-L327